### ISSUE TITLE
Ansible provisioner "inventory" option

### ISSUE THREAD

**Issue by jchampion** [lxdock/lxdock#88]
Il would be nice to have an option in Ansible provisioner that would allow us to force the inventory path. The inventory that is generated right now is very simplistic and does not deal with groups. I guess you'd have to use only "all" hosts right now for it to work.

While generating the groups based on container names would be possible, the most simple solution is to allow providing the path to an inventory.

Example:
```yml
provisioning:
  - type: ansible
    playbook: deploy.yml
    inventory: inventory/dev
```

**Comment by hsoft** [comment]
@jchampion thanks for your PR! While I very much agree that the current situation is limiting, I'm not sure it's a good idea to go for this solution as a temporary workaround measure.

The problem here is that once we've added a new config parameter, we can't easily remove it without breaking old `lxdock.yml` config, so we'll be stuck with it for a while. If the only reason for its existence is a workaround to temporary limitations (I can't see other uses right now), then we'll give ourselves needless hassle.

Adding grouping to our dynamically generated ansible inventory file doesn't seem overwhelmingly difficult to add. We could add a new `groups` attribute to container elements, and then provisioners can use (or not) this information as they please.

What do you think?

**Comment by jchampion** [comment]
I would not see it as a temporary workaround but more like an advanced option.

Right now, ansible-playbook is executed with a generated inventory that looks like this:
```
X.X.X.X ansible_user=root
```
It does not contain groups, nor does it allow any kind of tweaking such as changing the SSH user/port or any of other ansible_* options (http://docs.ansible.com/ansible/intro_inventory.html#list-of-behavioral-inventory-parameters). That is why I think an "advanced" inventory option would not hurt.


As a sidenote was using only one container until now and I just realized that the provisioning is executed on each container.

Let's say I have an lxdock.yml file such as this one:
```yml
containers:
  - name: app
    hostnames:
      - app.loc
  - name: db
    hostnames:
      - db.loc
provisioning:
  - type: ansible
    playbook: provision.yml
```
ansible-playbook will be executed twice with the "simple" inventory mentioned above. Thing is, it is unlikely that "app" and "db" should be provisioned the same way...

I imagine one of these (or both) could be done:

Option 1
-----------
Generate an inventory that looks like this:
```
[app]
X.X.X.X ansible_user=root
```
ansible-playbook would be executed twice but the group would allow for a specific provisioning.

Option 2
-----------
Execute ansible-playbook only once with a generated inventory that looks like this:
```
[app]
X.X.X.X ansible_user=root

[db]
X.X.X.X ansible_user=root
```

What do you think ?

**Comment by hsoft** [comment]
> That is why I think an "advanced" inventory option would not hurt.

I agree 100%. LXDock is lacking in this department at the moment. My point was that pointing to a manually crafted inventory feels like a kludge. In a situation where LXDock has proper "advanced inventory" features, I don't see why would anyone manually craft an inventory file.

> As a sidenote was using only one container until now and I just realized that the provisioning is executed on each container.

Yes, I know. Personally, I run ansible manually in my multi-containers projects, with a manually crafted inventory file :)

That's also one of the improvements that is on my short term radar (the ability to run provisioning "globally").

## Options

Option 2 makes the most sense to me. It's the simplest solution because we only have to craft one inventory file. Then, when comes the time to run ansible, there's nothing stopping us, if we wish, to apply a host-based `--limit` to keep the previous behavior of applying provisioning individually to every container.

Thanks for your input @jchampion, we have joint interests and what you want, I pretty much want the same thing. I happen to have a little more free time these days, so we could do this together. You start with the inventory grouping part and I continue with the "global vs local provisioning"?

For the "host based variable override" part though, I would wait and see how the grouping pans out. I'm not sure yet if this kind of variable override would belong in the `containers` config or in the `provisioning` one. Grouping is already a grey area...

**Comment by jchampion** [comment]
> You start with the inventory grouping part and I continue with the "global vs local provisioning"?

I don't think it is possible to do the first without the second. We only have access to the info about one guest during provisioning right now (I think).
Do you intend on keeping backwards compatibility (one provisioning per container) ? Or switch completely to a global provisioning (including shell ?) ?

**Comment by hsoft** [comment]
I don't think that global provisioning applies well to shell and puppet. What I had in mind was simply to apply ansible provisioning once with a multi-machine inventory when it was declared outside of the `containers` section. For the other type of provisioning, the behavior would stay the same.

**Comment by hsoft** [comment]
@jchampion Here we go, #90

**Comment by hsoft** [comment]
Fixed in #90 and #95
